### ISSUE TITLE
Prevent too many calls to SkiaSharp's Canvas.Restore()

### ISSUE THREAD

**Issue by BrianAtZetica** [ScottPlot/ScottPlot#3523]
**Suggestion:** 

In SP5, the RenderManager calls SkiaSharp's Canvas.Save() before every RenderAction and Canvas.Restore() after every RenderAction.
However, it also calls Canvas.Restore a second time within the DisableClipping() method

```cs
        foreach (IRenderAction action in RenderActions)
        {
            sw.Restart();
            rp.Canvas.Save();
            action.Render(rp);
            rp.Canvas.Restore();
            rp.DisableClipping();
            actionTimes.Add((action.ToString() ?? string.Empty, sw.Elapsed));
        }
```

According to SkiaSharp method docstring, Restore should not be called more often than save. 

Suggestion is to use Canvas.SaveCount to prevent excessive callas to Canvas.Restore.

**Comment by swharden** [comment]
Related? https://github.com/ScottPlot/ScottPlot/pull/3517

**Comment by BrianAtZetica** [comment]
Note to Self:

Canvas.SaveCount never seems to drop below one, even if Restore is called too many times. I assume something under-the-hood is preventing this.

SkiaSharp.SKCanvas has this remark:

```CS
    // Remarks:
    //     This will equal the number of SkiaSharp.SKCanvas.Save calls minus SkiaSharp.SKCanvas.Restore
    //     calls + 1. The save count on a new canvas is 1.
   public int SaveCount => SkiaApi.sk_canvas_get_save_count(Handle);
```

Note also the following RenderActions that may need checking: 

- [ ] RenderDataBackground performs some clipping and calls DisableClipping without a canvas.save (so is unsafe). Should replace DisableClipping with an explicit save and restore call.
- [ ] RenderPlottables calls a save, and finishes with a call to DisableClipping (so is safe). But it doesn't always do a clip. DisableClipping should be replaced with an explicit call to canvas.Restore.
- [ ] HorizontalLine.RenderLast() callas DisableClipping without a canvas.save (so is unsafe).
- [ ] VerticalLine.RenderLast() callas DisableClipping without a canvas.save (so is unsafe).

StandardZoomRectangle.Render() is an example that looks most correct (save, clip, restore)

**Comment by swharden** [comment]
resolved in #3527
